sycl/include/sycl/ext/intel/esimd/math.hpp

```diff
@@ -93,56 +93,56 @@ saturate(simd<T1, SZ> src) {
 // abs
 namespace detail {
 
-template <typename T0, typename T1, int SZ>
-ESIMD_NODEBUG ESIMD_INLINE simd<T0, SZ>
-__esimd_abs_common_internal(simd<T1, SZ> src0) {
-  simd<T1, SZ> Result = simd<T0, SZ>(__esimd_abs<T1, SZ>(src0.data()));
-  return Result;
+template <typename TRes, typename TArg, int SZ>
+ESIMD_NODEBUG ESIMD_INLINE simd<TRes, SZ>
+__esimd_abs_common_internal(simd<TArg, SZ> src0) {
+  simd<TArg, SZ> Result = simd<TArg, SZ>(__esimd_abs<TArg, SZ>(src0.data()));
+  return convert<TRes>(Result);
 }
 
-template <typename T0, typename T1>
+template <typename TRes, typename TArg>
 ESIMD_NODEBUG
-    ESIMD_INLINE std::enable_if_t<detail::is_esimd_scalar<T0>::value &&
-                                      detail::is_esimd_scalar<T1>::value,
-                                  std::remove_const_t<T0>>
-    __esimd_abs_common_internal(T1 src0) {
-  using TT0 = std::remove_const_t<T0>;
-  using TT1 = std::remove_const_t<T1>;
-
-  simd<TT1, 1> Src0 = src0;
-  simd<TT0, 1> Result = __esimd_abs_common_internal<TT0>(Src0);
-  return Result[0];
+    ESIMD_INLINE std::enable_if_t<detail::is_esimd_scalar<TRes>::value &&
+                                      detail::is_esimd_scalar<TArg>::value,
+                                  std::remove_const_t<TRes>>
```

Comment:
BTW, non-blocking comment: this remove_const_t<TRes> looks odd to me (it is not something added by your change). 
IMO it should be eventually removed.
If user specified TRes as 'const int', then why return 'int'? Also, what if TRes = 'const int *', then returning return 'int *' doesn't seem right.